### ISSUE TITLE
SystemError: new style getargs format but argument is not a tuple

### ISSUE THREAD

**Issue by spaceexperiment** [python-pillow/Pillow#2802]
I am trying to generate thumb from png images, but i get this error sometime.
it's fixed by opening the image in photoshop and saving it again.

i am using sorl.thumbnail, but the error seems to be from PIL

```python
im = get_thumbnail(p.image, '100x100', crop='center')
```

here is the error i am getting

```python
 File "/.../lib/python3.5/site-packages/sorl/thumbnail/engines/pil_engine.py", line 103, in _colorspace
    mask = Image.new('L', image.size, color=transparency)
  File "/.../lib/python3.5/site-packages/PIL/Image.py", line 2287, in new
    return Image()._new(core.fill(mode, size, color))
SystemError: new style getargs format but argument is not a tuple
```

**Comment by wiredfool** [comment]
You're passing something in for color, and it's not getting interpreted correctly.  Given that you're passing in 'transparency', you should make sure that it's a valid colorspec. I'd suggest that it should be a single int, 0<= color <=255. (while there are other options, transparency is not likely to be a string with a css color, e.g. "white")

**Comment by homm** [comment]
@spaceexperiment Which version of Pillow do you use?

**Comment by spaceexperiment** [comment]
@homm I am using 4.2.1, tried 4.3.0, still the same.
I was able to make things work by converting all my images to PNG and convert them to mode 'RGBA' , i noticed for example the issue was when for example the image PNG and the color mode was 'P'. converting  it to 'RGBA' fixed it.

I am assuming sorl.thumbnail is passing the transparency color for all image even if the image don't support it ?, but i am  not really sure.

I opened the same issue here [https://github.com/jazzband/sorl-thumbnail/issues/521](https://github.com/jazzband/sorl-thumbnail/issues/521)
